src/bun.js/webcore/streams.zig

```diff
@@ -1021,6 +1023,11 @@ pub const StreamResult = union(Tag) {
                 const value = result.toJS(globalThis);
                 value.ensureStillAlive();
 
+                switch (result.*) {
+                    .temporary => |*temporary| temporary.deinitWithAllocator(bun.default_allocator),
+                    .temporary_and_done => |*temporary_and_done| temporary_and_done.deinitWithAllocator(bun.default_allocator),
+                    else => {},
+                }
                 result.* = .{ .temporary = .{} };
```

Comment:
Will this cause JSValues for `.owned`, etc. variants to point to leaked/undefined memory?

Reply:
Does line 1046 move or borrow?

Man I miss Rust's borrow checker...